src/ImageSharp/Processing/Processors/CloningImageProcessor{TPixel}.cs

```diff
@@ -144,5 +153,17 @@ protected virtual void AfterFrameApply(ImageFrame<TPixel> source, ImageFrame<TPi
         protected virtual void AfterImageApply(Image<TPixel> destination)
         {
         }
+
+        /// <summary>
+        /// Disposes the object and frees resources for the Garbage Collector.
+        /// </summary>
+        /// <param name="disposing">Whether to dispose managed and unmanaged objects.</param>
+        protected virtual void Dispose(bool disposing)
```

Comment:
If we expect unmanaged objects (at least in theory) we should consider adding a finalizer.

In `Image`, I have distinnguished the dispose implementation by naming our method [`DisposeImpl`](https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp/Image.cs#L149). Still wondering if it's a good approach or we should rather `Dispose(bool disposing)`, even if the `bool` parameter is dummy.

Reply:
I'd say stick with convention and use `Dispose(bool disposing)`

Reply:
I didn't use a finalizer since I didn't want to use (or condone) unmanaged object use.